internal/util/lock/lock_util.go

```diff
@@ -0,0 +1,47 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lock
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/milvus-io/milvus/internal/util/logutil"
+	"github.com/milvus-io/milvus/internal/util/retry"
+	"go.uber.org/zap"
+)
+
+func TryLock(ctx context.Context, source string, mutex *sync.RWMutex,
+	interval time.Duration, retryCount uint, printDuration bool) bool {
+	log := logutil.Logger(ctx).With(zap.String("source", source))
+	beforeTryLock := time.Now()
+	err := retry.Do(ctx, func() error {
```

Comment:
Try lock should fail while ctx  receive done

Reply:
retry can cover context cancel case